source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -3317,4 +3385,135 @@ export class TablePeer extends TypedCardElementPeer<Adaptive.Table> {
+                        let page = new Adaptive.CarouselPage();
+
+                        (this.cardElement as Adaptive.Carousel).addPage(page);
+
+                        this.updateChildren();
+
+                        // We've added a new carousel page, so the carousel peers need a layout update
+                        this.designerSurface.needsLayoutUpdate = true;
+                    }
+                }
+            )
+        )
+    }
+
+    initializeCardElement() {
+        super.initializeCardElement();
+
+        (this.cardElement as Adaptive.Carousel).addPage(new Adaptive.CarouselPage());
+
+        // After initializing, we'll need to update the layout
+        this.designerSurface.needsLayoutUpdate = true;
+    }
+
+    populatePropertySheet(propertySheet: PropertySheet, defaultCategory: string = PropertySheetCategory.DefaultCategory) {
+        super.populatePropertySheet(propertySheet, defaultCategory);
+
+        propertySheet.remove(CardElementPeer.isVisibleProperty,
+            ContainerPeer.bleedProperty, 
+            ContainerPeer.styleProperty);
+
+        propertySheet.add(
+            defaultCategory,
+            CarouselPeer.timerProperty);
+    }
+
+    canDrop(peer: DesignerPeer) {
+        return false;
+    }
+
+    isDraggable(): boolean {
+        return false;
+    }
+
+    attachOnPageChange() {
+        (this.cardElement as Adaptive.Carousel).onPageChanged = (activeIndex: number, realIndex: number) => {
+            const carouselElement = this.cardElement as Adaptive.Carousel;
+
+            // TODO: I think this is stopping the peers from updating every time/when we click the arrows and there's only one page
+            if (carouselElement.currentIndex !== realIndex) {
+                if (activeIndex === 0) {
+                    // Index 0 is a duplicate slide, and we should slide to the end
+                    carouselElement.carousel?.slideTo(this.getChildCount())
+                } else if (activeIndex === (this.getChildCount() + 1)) {
+                    // Last index is a duplicate slide, and we should slide to the beginning
+                    carouselElement.carousel?.slideTo(1);
+                } else {
+                    // Valid index, rerender the card
+                    carouselElement.currentIndex = realIndex;
+                    this.designerSurface.render();
+                }
+
+                // Since the carousel changed, we will need to update the designer surface layout
+                this.designerSurface.needsLayoutUpdate = true;
+            }
+        };
+    }
+}
+
+export class CarouselPagePeer extends ContainerPeer {
+
+    isDraggable(): boolean {
+        return false;
+    }
+
+    getBoundingRect(): Rect {
+        const boundingRect = super.getBoundingRect();
+
+        const carousel = this.cardElement.parent as Adaptive.Carousel;
+
+        if (carousel && carousel.carouselPageContainer) {
```

Comment:
perhaps a little simpler

```suggestion
        if (carousel?.carouselPageContainer) {
```